eclipse-collections-code-generator/src/main/resources/api/map/mutablePrimitivePrimitiveMap.stg

```diff
@@ -102,6 +102,27 @@ public interface Mutable<name1><name2>Map extends <name1><name2>Map, Mutable<nam
      */
     <type2> getIfAbsentPut(<type1> key, <type2> value);
 
+    /**
+     * Retrieves the value associated with the key if one exists;
+     * associates a putValue with the key.
+     * @param key the key
+     * @param putValue the value to associate with {@code key} if no such mapping exists
+     * @param defaultValue the value to return if no mapping associated with {@code key} exists
+     * @return the value associated with key, if one exists, or {@code defaultValue} if not
+     * @since 11.1.
+     */
+    default <type2> getAndPut(<type1> key, <type2> putValue, <type2> defaultValue)
+    {
```

Comment:
There are two ways to simplify this:

```
       <type2> returnValue = defaultValue;
        if (this.containsKey(key))
        {
            returnValue = this.get(key);
        }
        this.put(key, putValue);
        return returnValue;
```

or:

```
        if (this.containsKey(key))
        {
            defaultValue = this.get(key);
        }
        this.put(key, putValue);
        return defaultValue;
```

Reply:
I think the whole point of this method is so we don't look through the map three times. I guess you could simplify it to 2, with:
```
returnValue = this.getIfAbsent(key, defaultValue);
this.put(key, putValue);
return returnValue;
```
but it's not too egregious to copy the put method. It's well factored and some of the methods can be repurposed.
```
    public void putAndGet(int key, int putValue, int defaultValue)
    {
        if (isEmptyKey(key))
        {
            return this.putForEmptySentinel(value, defaultValue); // change this method to return the old value if any
        }

        if (isRemovedKey(key))
        {
            return this.putForRemovedSentinel(value, ); // change this method to return the old value if any
        }

        int index = this.probe(key);
        int keyAtIndex = this.keysValues[index];
        if (keyAtIndex == key)
        {
            int returnValue = this.keyValues[index + 1];
            this.keysValues[index + 1] = value;
            return returnValue;
        }
        else
        {
            this.addKeyValueAtIndex(key, value, index);
            return defaultValue; //because the key wasn't in the map
        }
    }
```

Reply:
Hi @mohrezaei, this is for the default method on the interface. Our options were to throw UOE here or have a slightly suboptimal default implementation for anyone who happens to implement this interface and doesn't override. Your `getIfAbsent`() usage is better here.

Reply:
fixed. thanks.